app/src/components/ArtifactTimeline.tsx

```diff
@@ -0,0 +1,112 @@
+import { Center, ScrollArea, Space, Text, Timeline } from '@mantine/core';
+import { Message } from 'tabler-icons-react';
+import { useEffect, useState } from 'react';
+import ArtifactComments from './ArtifactComments';
+
+interface ArtifactCommentProps {
+  date: string | undefined;
+  body: string | undefined;
+  author: string | undefined;
+}
+
+interface ExtensionArray {
+  extensions: undefined | fhir4.Extension[];
+}
```

Comment:
```suggestion
interface ArtifactTimelineProps {
  extensions?: fhir4.Extension[];
}
```
I think this name makes a little more sense and you can do the `?` in place of `undefined` again!